packages/protocol/contracts/stability/Reserve.sol

```diff
@@ -302,10 +311,45 @@ contract Reserve is
    * @param spender The address that is to be no longer allowed to spend Reserve funds.
    */
   function removeSpender(address spender) external onlyOwner {
-    isSpender[spender] = false;
+    delete isSpender[spender];
     emit SpenderRemoved(spender);
   }
 
+  /**
+   * @notice Gives an address permission to spend Reserve without limit.
+   * @param spender The address that is allowed to spend Reserve funds.
+   */
+  function addExchangeSpender(address spender) external onlyOwner {
+    require(!isExchangeSpender.isSpender[spender], "Address is already Exchange Spender");
+    isExchangeSpender.isSpender[spender] = true;
+    // isExchangeSpender.addresses.push(spender);
+    isExchangeSpender.addresses.length = isExchangeSpender.addresses.push(spender);
+    emit ExchangeSpenderAdded(spender);
+    // TODO shall I add to the lenght?
+  }
+
+  /**
+   * @notice Takes away an address's permission to spend Reserve funds without limits.
+   * @param spender The address that is to be no longer allowed to spend Reserve funds.
+   */
+  function removeExchangeSpender(address spender, uint256 index) external onlyOwner {
+    isExchangeSpender.isSpender[spender] = false;
+    uint256 numAddresses = isExchangeSpender.addresses.length;
+    require(index < numAddresses, "Index is invalid");
+    require(spender == isExchangeSpender.addresses[index], "Index does not match spender");
+    uint256 newnumAddresses = numAddresses.sub(1);
```

Comment:
I'd probably follow https://github.com/celo-org/celo-monorepo/blob/master/packages/protocol/contracts/identity/Attestations.sol#L300 here. I.e. no need for the comments, seting the length, but I believe you do need to zero out the last element

Reply:
That's where I got a little list, because there we are indeed setting the length: https://github.com/celo-org/celo-monorepo/blob/master/packages/protocol/contracts/identity/Attestations.sol#L313

Reply:
and it doesn't seem to need to replace the `0x0` as the test pass with the right amount.

Reply:
I believe it is still good practice to do in case the future has some buggy implementation that would allow the extension of the length

Reply:
Btw, slightly incorrect camelization => `newnum` => `newNum`